utils/FileLoader.h

```diff
@@ -0,0 +1,73 @@
+#ifndef __FILELOADER_H__
+#define __FILELOADER_H__
+
+#include <stdtype.h>
+#include <stdio.h>
+#include <zlib.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+union LoaderHandles
+{
+	const void *dataPtr;
+    FILE *hFileRaw;
+    gzFile hFileGZ;
+};
+
+typedef union LoaderHandles LoaderHandles;
+
+enum
+{
+	FLSTAT_EMPTY = 0,
+	FLSTAT_LOADING = 1,
+	FLSTAT_LOADED = 2
+};
+
+enum
+{
+	// mode: sources
+	FLMODE_SRC_MEMORY = 0x01,
+	FLMODE_SRC_FILE = 0x02,
+
+	// mode: compression
+	FLMODE_CMP_RAW = 0x00,
+	FLMODE_CMP_GZ = 0x10
+};
+
+struct FileLoader {
+	UINT8 _modeSrc;
+	UINT8 _modeCompr;
+	UINT8 _status;
+	UINT32 _bytesTotal;
+	UINT32 _bytesLoaded;
+    UINT32 _readStopOfs;
+    LoaderHandles _hLoad[1];
+    UINT8 *_data;
+};
+
+typedef struct FileLoader FileLoader;
```

Comment:
Please change the typedef's name to FILE_LOADER, so that we're staying consistent with the rest of the coding style.